File: show-case-ui/src/domain/person.ts
export interface Person {
  id: string;
  firstName: string;
  lastName: string;
  birthdate: Date;
  city: string;
}

export function PersonArrayJSON(json: any): Person[] {
  return json.map((person: any): Person => {
    return {
      id: person["id"],
      firstName: person["firstName"],
      lastName: person["lastName"],
      birthdate: person["birthdate"],
      city: person["city"],
    };
  });
}

File: person-management-service/src/main/java/ch/fn/personmanagementservice/domain/Person.java
package ch.fn.personmanagementservice.domain;

import java.time.LocalDate;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Person {

  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  private Long id;

  private String firstName;
  private String lastName;
  private LocalDate birthdate;
  private String city;
}
